spec/horoshop/connection_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Horoshop::Connection do
+  subject { object.new('http://api.horosop.com') }
+
+  let(:object) do
+    Class.new do
+      include Horoshop::Connection
+      attr_accessor :url
+
+      def initialize(url)
+        @url = url
+      end
+    end
+  end
+
+  describe '#post' do
+    let(:url) { '/api/test' }
+    let(:body) { { key: 'value' } }
+
+    context 'when request is successful' do
+      before do
+        stub_request(:post, "http://api.horosop.com#{url}")
+          .to_return(status: 200, body: '{"status":"OK","response": { "token": "some_token"} }')
+      end
+
+      it 'returns a success response' do
+        response = subject.post(horoshop: subject, url: url, body: body)
+        expect(JSON.parse(response.body)).to include('status' => 'OK', 'response' => { 'token' => 'some_token' })
+      end
+    end
+
+    context 'when a server error occurs' do
+      before do
+        stub_request(:post, "http://api.horosop.com#{url}")
+          .to_return(status: 500, body: '')
```

Comment:
body will be nil when you receive 500 error

Reply:
No, because the faraday::error interceptor will fire and the response status will be 500, after which we add the body of the ERROR constant